server/lib/src/filter.rs

```diff
@@ -1287,8 +1287,7 @@ impl FilterResolved {
 
                 // If the f_list_or only has one element, pop it and return.
                 if f_list_new.len() == 1 {
-                    #[allow(clippy::expect_used)]
-                    f_list_new.pop().expect("corrupt?")
+                    f_list_new.remove(0)
```

Comment:
This should also be a [clear](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.clear)
```suggestion
                    f_list_new.clear()
```

Reply:
Uhhh? no? clear just deletes everythdng. We want to *pop* the last element and return it.

Reply:
Derp I r dumb. All good 😄